### ISSUE TITLE
Better implemenatation to display any menu item's default value

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#1346]
I think our implementation of showing a menu's default value is flawed.

The standard solution would be something like this: possible values are e.g. 11,12,13,14 (default),15,16, where, as can be seen, the default value would be indicated by a trailing " (default)", properly localized, of course.

Instead, we use "Default" (with no indication what the value is) in a menu's line item display, and in the selection dialogue for the value itself e seem to have no indication at all what the default value is.

**Comment by vshcherb** [comment]
It is all configurable via rendering style, I don't really see a programming issue here

**Comment by sonora** [comment]
True, it's probably easy to fix, but it is not fixed as of now. I think we should implement this (I see at least 3 places where we have the behavior (only recently introduced) I said needs improvement: Contour line zoom, Hiking symbol overlay, and Road style. Not sure if there are more. Leonid, I guess we only need to go through the renderers and check for occurrence of "defaultValueDescription" ?

**Comment by sonora** [comment]
I have looked at it again and think we should change things in our code. This is something which should not be configurable in the rendering style at all, because it is fully deterministic: We need to
(1) For any menu (where we want this) find out what our default value is (this Is specified in our settings definitions/activity)
(2) In the menu itself add " (default)" behind the respective display entry, (where "default" of course needs to be properly localized)
(3) When the users picks this very entry in the menu, we may need to strip the " (default)" so we pass only the actual menu entry value if our code expects this

Our current implementation does none of this, instead it produces the following useless behavior, one example:
(a) Fresh start of OsmAnd, go to Map Configuration, look at "Show contour lines": The menu item will show "Default" on the right hand side (where the selected value is supposed to show up)
(b) Open the "show contour lines" dialogue and select e.g. 11 (which is _not_ our default)
(c) Kill OsmAnd and restart. Do (a). Menu will again show "Default" (which is wrong). Opening the dialogue will correctly display 11.
 

What I am saying is we should either have no default indicator in our menus or have a proper implementation of how it is conventionally displayed. What we have now is flawed or useless.

**Comment by vshcherb** [comment]
My comments: 
- Default value is always 1 and it is always passing to rendering style as "" (empty string). For UI there is a special entry defaultValueDescription to 
- I don't really think there is value to see which value is default, it takes too much space (in some languages it makes dialogs too big) and by convention it is always first entry in Dialog. It could be configured in style to add additional meaning if needed, like Contour Lines - 13 (Optimal).
 - In Map Configuration of course it should be consistent with what you see in dialog, so it should be either valueDescription or defaultValueDescription depending what is selected. 

So far I see only one check needs to be done with Map Configuration.

**Comment by sonora** [comment]
So guess we are saying the same things:
(1) An indication of which value is the default may be nice to have, but is maybe too awkward in many languages so I can agree let's rather not have it.
(2) But then what purpose does our parameter defaultValueDescription serve, because as it is implemented now it simply generically displays "Default" (or the localized translation of it), and when displayed it gives no indication what value it has
(3) Plus: We still have a bug here: If I select any value, after restarting OsmAnd the menu will always display "Default", even if I selected a value which is clearly not the default. (I tested with contour line minzoom, but think it is the same for other parameters as well)

Sounds to me like we should simply back out our implementation around defaultValueDescription, or does it serve any useful purpose?
